semgrep_output_v1.atd

```diff
@@ -169,7 +169,15 @@ type core_match <python decorator="dataclass(frozen=True)"> = {
 
 (* TODO: try to make it as close as possible to 'cli_match_extra' below *)
 type core_match_extra <python decorator="dataclass(frozen=True)"> = {
-  ?message: string option; (* rule.message (?) *)
+  (* These fields generally come from the rule, but may be set here if they're
+   * being overriden for that particular finding. This would currently occur
+   * for rule with a validator for secrets, depending on what the valdiator
+   * might match, but could be expanded in the future. *)
+  ?message: string option;
+  ?metadata: raw_json option;
+  (* LATER: use a variant for severity *)
+  ?severity: string option;
```

Comment:
let's clean this severity thing once and for all!
Let's define it just in the ATD and reuse it in semgrep-core, osemgrep, and pysemgrep.

Reply:
Happy to do this, but I'm not sure this solves everything. What's going on with Severity.ml in osemgrep? Should I get rid of that here too?

Reply:
I'll take care of Severity.ml

Reply:
@aryx Could you approve changes and the make the required changes for OSS develop, in that case? I'm not sure how you would want to handle it and would rather avoid trying to modify things at the same time.

Reply:
yes I can make the change to OSS.